### PR TITLE
Added Hilt as new Dependency Injection Manager

### PR DESCRIPTION
**Pull request by Filmaluco**
Pull request contains all the changes to support the new Android focused dependency manager [Hilt](https://dagger.dev/hilt/)

### CODE REVIEW

File: template/app/src/androidTest/java/io/bloco/template/ui/counter/CounterActivityTest.kt [hunk 1]
```diff
@@ -1,38 +1,67 @@
...
+            getInstrumentation().targetContext)
+        sharedPreferences.edit().clear().commit()
+        val flowSharedPreferences =  FlowSharedPreferences(sharedPreferences)
 
-    @get:Rule
+        runBlocking {
+            flowSharedPreferences.getInt(CounterRepository.COUNTER_KEY, defaultValue = 1).setAndCommit(1)
+        }
+        flowSharedPreferences
+    }
```
Comment (sdsantos) [inline template/app/src/androidTest/java/io/bloco/template/ui/counter/CounterActivityTest.kt:41]: 1) You already have a rule for clearing preferences. But you are also clearing them here?
2) Do you really need to replace with a new instance of FlowSharedPreferences? Can't you just inject here the existing FlowSharedPreferences instance provided, and just manipulate it? Or would that happen at the wrong time?

File: template/app/src/androidTest/java/io/bloco/template/ui/counter/CounterActivityTest.kt [hunk 2]
```diff
@@ -1,38 +1,67 @@
...
+
+    var hiltRule = HiltAndroidRule(this)
     val intentTestRule = ActivityScenarioRule<CounterActivity>(CounterActivity::class.java)
 
+    @get:Rule
+    var rules = RuleChain.outerRule(hiltRule)
+        .around(intentTestRule)
```
Comment (sdsantos) [inline template/app/src/androidTest/java/io/bloco/template/ui/counter/CounterActivityTest.kt:48]: If they don't provide a shorter way to do this in the next versions, we may want to make our own rule that ties together both.

File: template/app/src/androidTest/java/io/bloco/template/ui/counter/CounterActivityTest.kt (hunk 1 above)
Comment (sdsantos) [reply template/app/src/androidTest/java/io/bloco/template/ui/counter/CounterActivityTest.kt:41]: 3. You can also just inject the GetAndSetCounter and use it to set a value.

**Review by sdsantos — APPROVED** [review]

File: template/app/src/androidTest/java/io/bloco/template/ui/counter/CounterActivityTest.kt (hunk 1 above)
Comment (Filmaluco) [reply template/app/src/androidTest/java/io/bloco/template/ui/counter/CounterActivityTest.kt:41]: about 1, I no longer have the rule I deleted it.
not sure if I can do 2, need to check it in more detail
3rd I can do, but the "idea" was to have hilt here as kinda of a basic example.

Comment (sdsantos) [reply template/app/src/androidTest/java/io/bloco/template/ui/counter/CounterActivityTest.kt:41]: You still need Hilt to inject `GetAndSetCounter` into the test ;)

**Pull request merged**